### DISCUSSION TITLE
How to detect when a user clicks a tab in a TabbedPage

### DISCUSSION THREAD

**Discussion by rogitha** [dotnet/maui#25721] [Q&A]
To preface, I am a college student working on a mobile project for a class using .NET MAUI. Everyone in the class is contributing towards the project, so it is rather large. I am not comfortable making to large of changes as to not impact other groups. Our main page consists of a TabbedPage whose children are NavigationPages that each group essentially works on. 

Currently, my group's page will navigate to other pages (not on the tab bar) by using Navigation.pushAsync() and buttons. What we want to happen, when the user clicks the tab for our page (while they are currently viewing it), is for our page to go back to the initial screen with all the buttons using Navigation.PopToRootAsync(). Essentially, we want our page to refresh when the tab button is clicked. 

From what I can find, TabbedPage has an event for when the current page has changed, TabbedPage.CurrentPageChanged. However, this does not see to register when a user clicks the same tab twice. This make sense because the current page hasn't actually changed, but there doesn't seem to be anyway to detect if the user has clicked that same tab again. 

How can I tell when a user has clicked a tab in the TabbedPage, or detect when the user has clicked the same tab twice or more?

**Comment by LeoJHarris** [comment]
This might be what your after:
 https://github.com/dotnet/maui/issues/17877#issuecomment-2486681222

Achieved through the use of a handler.
